### PR TITLE
chore: Update lints

### PR DESCRIPTION
**Pull request by dnys1**
- Updates `amplify_lints` to use `package:lints` and `package:flutter_lints` v2.0
- Consolidates rulesets into `library` and `app`
- Renames current ruleset as the "core" rulesets for libraries/apps
- Expands current rulesets to include more lints. Packages can be migrated incrementally to the new ruleset.
- Fixes resulting lint violations of migrating a couple packages to the new rulesets.

### CODE REVIEW

**Review by Jordan-Nelson — APPROVED** [review]

File: packages/amplify_core/android/src/test/kotlin/com/amazonaws/amplify/amplify_core/AtomicResultTest.kt [hunk 1]
```diff
@@ -15,10 +15,8 @@
 package com.amazonaws.amplify.amplify_core
 
 import io.flutter.plugin.common.MethodChannel
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.Job
-import kotlinx.coroutines.launch
+import kotlinx.coroutines.*
```
Comment (HuiSF) [inline packages/amplify_core/android/src/test/kotlin/com/amazonaws/amplify/amplify_core/AtomicResultTest.kt:18]: I thought the codebase is against wildcards importing.

File: packages/amplify_core/lib/src/types/models/auth_rule.dart [hunk 2]
```diff
@@ -84,24 +84,8 @@ class AuthRule {
       ..removeWhere((k, dynamic v) => v == null);
   }
 
-  factory AuthRule.fromMap(Map<String, dynamic> map) {
```
Comment (HuiSF) [inline packages/amplify_core/lib/src/types/models/auth_rule.dart:87]: Did the linter suggest to remove these two factories as they are not used or something?

File: packages/amplify_core/android/src/test/kotlin/com/amazonaws/amplify/amplify_core/AtomicResultTest.kt (hunk 1 above)
Comment (dnys1) [reply packages/amplify_core/android/src/test/kotlin/com/amazonaws/amplify/amplify_core/AtomicResultTest.kt:18]: Hmm I typically just let AS figure it out. Is there a setting to disable it?

File: packages/amplify_core/lib/src/types/models/auth_rule.dart (hunk 2 above)
Comment (dnys1) [reply packages/amplify_core/lib/src/types/models/auth_rule.dart:87]: They aren't being used in the codebase and threw a bunch of lint errors. I started fixing them but realized some were just incorrect so I gave up and just removed em.

**Comment by dnys1** [comment]
✓ Commit merged in pull request [#1620](https://github.com/aws-amplify/amplify-flutter/pull/1620)

**Pull request closed without merging**